paddle/phi/kernels/onednn/reduce_kernel.h

```diff
@@ -0,0 +1,150 @@
+/* Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```

Comment:
`xxx_kernel.h` has a specific meaning. We only place the kernel declaration in `xxx_kernel.h`. It is recommended to replace it with another name to avoid misunderstanding, such as `reduce_kernel_utils.h`